exporter/loadbalancingexporter/trace_exporter.go

```diff
@@ -207,20 +109,19 @@ func (e *traceExporterImp) consumeTrace(ctx context.Context, td pdata.Traces) er
 		return errNoTracesInBatch
 	}
 
-	// NOTE: make rolling updates of next tier of collectors work. currently this may cause
-	// data loss because the latest batches sent to outdated backend will never find their way out.
-	// for details: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1690
-	e.updateLock.RLock()
-	endpoint := e.ring.endpointFor(traceID)
-	exp, found := e.exporters[endpoint]
-	e.updateLock.RUnlock()
-	if !found {
-		// something is really wrong... how come we couldn't find the exporter??
-		return fmt.Errorf("couldn't find the exporter for the endpoint %q", endpoint)
+	endpoint := e.loadBalancer.Endpoint(traceID)
+	exp, err := e.loadBalancer.Exporter(endpoint)
+	if err != nil {
+		return err
+	}
+
+	te, ok := exp.(component.TracesExporter)
+	if !ok {
+		return errUnexpectedExporterType
```

Comment:
Be nice to the caller and tell them which type it was :-)

Reply:
Fixed, in the latest commit. Had to rebase due to conflict with main.